lib/ferver/hash_cache.rb

```diff
@@ -0,0 +1,31 @@
+module HashCache
+  require 'yaml/store'
+
+  ## 
+  #  An ETag cache.
+  #
+  #  Given a file size, mtime and inode, reply with the previous sha1 hash
+  ##
+  
+  def key(file)
+    f = File.stat(file) 
+    inode = f.ino
+    mtime = f.mtime.to_i
+    size  = f.size
+    "#{inode}#{mtime}#{size}"
+  end
+
+  def hash(file)
+    store = YAML::Store.new( File.join(ENV['HOME'], ".ferver.yml"))
```

Comment:
Space inside parentheses detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.